polaris/polaris/management/commands/watch_transactions.py

```diff
@@ -18,144 +21,162 @@
+        """
+        try:
+            memo_type = response["memo_type"]
+            response_memo = response["memo"]
+            successful = response["successful"]
+            stellar_transaction_id = response["id"]
+            envelope_xdr = response["envelope_xdr"]
+        except KeyError:
+            return False
+
+        if memo_type != "hash":
+            return False
+
+        # The memo on the response will be base 64 string, due to XDR, while
+        # the memo parameter is base 16. Thus, we convert the parameter
+        # from hex to base 64, and then to a string without trailing whitespace.
+        if response_memo != format_memo_horizon(transaction.withdraw_memo):
+            return False
+
+        horizon_tx = TransactionEnvelope.from_xdr(
+            response["envelope_xdr"],
+            network_passphrase=settings.STELLAR_NETWORK_PASSPHRASE,
+        ).transaction
+        found_matching_payment_op = False
+        for operation in horizon_tx.operations:
+            if self._check_payment_op(
+                operation, transaction.asset.code, transaction.amount_in
+            ):
+                found_matching_payment_op = True
+                break
+
+        return found_matching_payment_op
+
+    @staticmethod
+    def update_transaction(
+        response: Dict, transaction: Transaction, error_msg: str = None
+    ):
+        """
+        Updates the transaction depending on whether or not the transaction was
+        successfully executed on the Stellar network and `process_withdrawal`
+        completed without raising an exception.
+
+        If the Horizon response indicates the response was not successful or an
+        exception was raised while processing the withdrawal, we mark the status
+        as `error`. If the Stellar transaction succeeded, we mark it as `completed`.
+
+        :param error_msg: a description of the error that has occurred.
+        :param response: a response body returned from Horizon for the transaction
+        :param transaction: a database model object representing the transaction
+        """
+        if error_msg or not response["successful"]:
+            transaction.status = Transaction.STATUS.error
+            transaction.status_message = error_msg
+        else:
+            transaction.completed_at = now()
+            transaction.status = Transaction.STATUS.completed
+            transaction.status_eta = 0
+            transaction.amount_out = transaction.amount_in - transaction.amount_fee
+
+        transaction.stellar_transaction_id = response["id"]
+        transaction.save()
+
+    @staticmethod
+    def _check_payment_op(
+        operation: Operation, want_asset: str, want_amount: float
+    ) -> bool:
+        return (
+            operation.type_code() == Xdr.const.PAYMENT
+            and str(operation.destination)
+            == settings.STELLAR_DISTRIBUTION_ACCOUNT_ADDRESS
```

Comment:
it looks like we no longer create `STELLAR_DISTRIBUTION_ACCOUNT_ADDRESS` how does this work?

Reply:
Good catch, it wouldn't work.

Reply:
do we need a test case to cover this?

Reply:
Yea it should have one, I'll add it.